### PR TITLE
[WIP] insert tweet

### PR DESCRIPTION
**Pull request by jaljo**

### CODE REVIEW

File: text-editor/text-editor-003-en.md [hunk 1]
```diff
@@ -1 +1,102 @@
 ## Insertion of a tweet : a case study
+
+In this chapter, we will try to give a better insight of how the whole thing is
+working, by breaking down the insertion of a tweet case and the components
+involved in it's workflow.
+
+### Getting the HTML markup from twitter
```
Comment (nicolasmure) [inline text-editor/text-editor-003-en.md:8]: I think it'd be nice to describe the behavior of this feature first : be able to insert a rendered tweet into the edited text, just by pasting the URL of the tweet (ie no more handmade HTML edition to insert an embed of a tweet).

File: text-editor/text-editor-003-en.md [hunk 2]
```diff
@@ -1 +1,102 @@
...
+
+Keep in mind that our main goal here is to obtain the simplest possible HTML
+representation of our tweet, so it can lately be saved to the API. That way it
+will be easy for us to parse it as a component, and finally display it to the
+end user in the front end application.
+
+The following chart will give you an overview of the actions flow :
+![Insert tweet action flow chart](https://github.com/jaljo/articles/raw/314724209eb27b7b1d61795341934c4783084bee/images/text-editor/RTE.png)
```
Comment (nicolasmure) [inline text-editor/text-editor-003-en.md]: You can directly set the path to `/images/text-editor/RTE.png` so it'll always display the version of this file corresponding to the version of the `.md` file (same revision)

File: text-editor/text-editor-003-en.md [hunk 3]
```diff
@@ -1 +1,102 @@
...
+your fellow developpers dont want to break your legs when the prod crashes
+(which will happend, but that's another story).
+
+### Inserting the tweet into the DOM
+
+That last action is observed by another epic which will perform the actual tweet
+insertion into the DOM (as epic are meant to deal with side effects, it's the
+proper place in the project to do that). How is that ?
+
+[See gist here](https://gist.github.com/jaljo/4c8f83acc48766b930d7d584fe9ed22b)
```
Comment (nicolasmure) [inline text-editor/text-editor-003-en.md:100]: To improve readability of this file, I think it'd be good to put the actual code in this file too. A link to the gist can be placed below the code ;)

File: text-editor/text-editor-003-en.md [hunk 4]
```diff
@@ -1 +1,102 @@
...
+So the first epic is only responsible to call that endpoint and dispatch a
+`EMBED_TWEET_FETCHED` action that carries out the obtained HTML markup in its
+payload. Its always convenient to isolate API calls in dedicated epics so errors
+can be properly catched and displayed to the end user when something goes wrong.
+Cool thing is, keeping epics as atomic as possible (observe an action, do one
+thing, then dispatch another action) allow them to be easily tested. That is,
+your fellow developpers dont want to break your legs when the prod crashes
+(which will happend, but that's another story).
+
+### Inserting the tweet into the DOM
```
Comment (nicolasmure) [inline text-editor/text-editor-003-en.md]: I think it's be nice to say that we're inserting the `embed` code here, not the full rendered tweet.

File: text-editor/text-editor-003-en.md [hunk 5]
```diff
@@ -1 +1,102 @@
...
+as a consequence of the previous action: a new paragraph has been inserted),
+- One that will close the tweet insertion form so that when the paragraph toolbox
+is lately shown the user is not annoyed,
+- The last wich only maps the payload of the `TWEET_INSERTED` action payload to
+a new `RENDER_TWEET` one. Although it can sound a bit overkill to have an epic
+that does apparently 'nothing', it help us keep our action stream crystal clear.
+
+Guess what ? That action is observed too !
+
+[See gist here](https://gist.github.com/jaljo/006cab23716b9076dfcd5b2032d03081)
```
Comment (nicolasmure) [inline text-editor/text-editor-003-en.md:157]: same here : can you put the code above this gist link please ?

**Review by nicolasmure — COMMENTED** [review]
Nice !! :bird:

File: text-editor/text-editor-003-en.md (hunk 2 above)
Comment (jaljo) [reply text-editor/text-editor-003-en.md]: Nice trick, I was not aware of that :+1:

**Review by nicolasmure — APPROVED** [review]
thanks !! :sunglasses:

**Pull request merged**